machine/fdt.c

```diff
@@ -204,6 +204,64 @@ void query_mem(uintptr_t fdt)
   assert (mem_size > 0);
 }
 
+///////////////////////////////////////////// ROOT SCAN /////////////////////////////////////////
+
+struct root_scan {
+  const struct fdt_scan_node *root;
+  const uint32_t *version_value;
+  int version_len;
+};
+
+static void root_open(const struct fdt_scan_node *node, void *extra)
+{
+  struct root_scan *scan = (struct root_scan *)extra;
+  if (!scan->root) {
+    scan->root = node;
+  }
+}
+
+static void root_prop(const struct fdt_scan_prop *prop, void *extra)
+{
+  struct root_scan *scan = (struct root_scan *)extra;
+  if (!strcmp(prop->name, "sri-cambridge,version") && !scan->version_value) {
+    scan->version_value = prop->value;
+    scan->version_len = prop->len;
+  }
+}
+
+static int root_close(const struct fdt_scan_node *node, void *extra)
+{
+  struct root_scan *scan = (struct root_scan *)extra;
+  if (scan->root && node == scan->root && scan->version_value) {
```

Comment:
```suggestion
  if (scan->root == node && scan->version_value) {
```
First condition is redundant given node isn't null, second one is an unusual way round

Reply:
Perhaps we should print "unknown" for these rather than nothing at all if not present?